File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.js
function solution(players, m, k) {
  let answer = 0;
  let server = Array(24).fill(0); // 시간대별 서버 개수

  for (let i = 0; i < 24; i++) {
    if (Math.floor(players[i] / m) > server[i]) {
      const adding = Math.floor(players[i] / m) - server[i];
      answer += adding;

      for (let j = i; j < i + k; j++) {
        if (j < 24) server[j] += adding;
      }
    }
  }

  return answer;
}

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
class Solution {

    public int solution(int[] players, int m, int k) {
        int answer = 0;
        int n = 0;
        
        int[] server = new int[24]; // 시간대별 서버의 수 
        
        for(int i=0; i<24; i++){
            
            if(players[i] / m > server[i])
            {
                int adding = (players[i] / m) - server[i];
                answer+= adding;
                
                for(int j=i; j<i+k; j++){
                    if(j < 24) server[j]+= adding;
                } 
            }
        }
        
        return answer;
    }
}